crates/burn-import/src/burn/node/constant_of_shape.rs

```diff
@@ -0,0 +1,185 @@
+use super::{Node, NodeCodegen};
+use crate::burn::{Scope, Type};
+use burn::record::PrecisionSettings;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+/// Node for all unary operators.
+#[derive(Debug, Clone)]
+pub struct ConstantOfShapeNode {
+    pub input: Type,
+    pub output: Type,
+    pub value: ConstantValue,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum ConstantValue {
+    /// Float constant.
+    Float32(f32),
+    Float64(f64),
+
+    /// Integer constant.
+    Int32(i32),
+    Int64(i64),
+
+    // Boolean constant.
+    Bool(bool),
+}
+
+impl ConstantOfShapeNode {
+    pub fn new(input: Type, output: Type, value: ConstantValue) -> Self {
+        assert!(
+            matches!(input, Type::Tensor(_)),
+            "ConstantOfShape input needs to be a Tensor!"
+        );
+        assert!(
+            matches!(output, Type::Tensor(_)),
+            "ConstantOfShape output needs to be a Tensor!"
+        );
+        Self {
+            input,
+            output,
+            value,
+        }
+    }
+}
+
+impl ConstantValue {
+    pub fn val_tokens(&self) -> TokenStream {
+        match self {
+            Self::Float32(val) => quote! { #val },
+            Self::Float64(val) => quote! { #val },
+            Self::Int32(val) => quote! { #val },
+            Self::Int64(val) => quote! { #val },
+            Self::Bool(val) => quote! { #val },
+        }
+    }
+
+    pub fn from_vec<T: Into<Self> + Copy>(mut source: Vec<T>) -> Self {
+        assert_eq!(
+            source.len(),
+            1,
+            "ConstantOfShape value from a vec needs to have exactly 1 element!"
+        );
+        source.drain(..).next().unwrap().into()
+    }
+}
+
+impl From<f32> for ConstantValue {
+    fn from(value: f32) -> Self {
+        Self::Float32(value)
+    }
+}
+impl From<f64> for ConstantValue {
+    fn from(value: f64) -> Self {
+        Self::Float64(value)
+    }
+}
+impl From<i32> for ConstantValue {
+    fn from(value: i32) -> Self {
+        Self::Int32(value)
+    }
+}
+impl From<i64> for ConstantValue {
+    fn from(value: i64) -> Self {
+        Self::Int64(value)
+    }
+}
+impl From<bool> for ConstantValue {
+    fn from(value: bool) -> Self {
+        Self::Bool(value)
+    }
+}
+
+impl<PS: PrecisionSettings> NodeCodegen<PS> for ConstantOfShapeNode {
+    fn input_types(&self) -> Vec<Type> {
+        vec![self.input.clone()]
+    }
+
+    fn output_types(&self) -> Vec<Type> {
+        vec![self.output.clone()]
+    }
+
+    fn forward(&self, _scope: &mut Scope, _node_position: usize) -> TokenStream {
+        let output = self.output.name();
+        let input = self.input.name();
+        let value = self.value.val_tokens();
+        // Note: in the generated code, self.device is a &module::Ignored<Device>, so to get a &Device, &* is needed:
+        quote! {
+            let #output = Tensor::full(#input.shape(), #value, &*self.device);
```

Comment:
According to [the spec](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ConstantOfShape), the input is a 1D tensor. So to convert to burn we probably need to do something like this instead:

```rust
let input = scope.tensor_use_owned(&input, node_position);
quote! {
    let #output = Tensor::full(#input.to_data().value, #value, &#input.device());
}
```

Also, the spec supports a constant scalar when no dimensions are given, so we would have to handle this particular use case. But in this PR we can always limit the scope to tensors for now.

Reply:
Fixed it by introducing shape type.